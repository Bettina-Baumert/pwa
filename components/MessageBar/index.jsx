/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';
import styles from './style';

/**
 * The message bar component.
 * @param {Object} props The component props.
 * @param {Array} props.messages The message content.
 * @return {JSX}
 */
const MessageBar = ({ messages }) => (
  <div className={styles.container}>
    {messages.map(({ type = 'info', message }) => (
      <div key={`${type}-${message}`} className={styles[type]}>
        {message}
      </div>
    ))}
  </div>
);

MessageBar.propTypes = {
  messages: PropTypes.arrayOf(
    PropTypes.shape({
      message: PropTypes.string,
      type: PropTypes.string,
    })
  ).isRequired,
};

MessageBar.defaultProps = {
  messages: [],
};

export default MessageBar;
