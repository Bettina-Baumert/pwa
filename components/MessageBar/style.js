/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { css } from 'glamor';
import colors from 'Styles/colors';
import variables from 'Styles/variables';

const container = css({
  display: 'flex',
  flexDirection: 'column',
  background: colors.background,
}).toString();

const messageBase = {
  padding: `${variables.gap.small}px ${variables.gap.big}px`,
  color: colors.light,
  fontSize: '0.875rem',
  fontWeight: 500,
  ':not(:last-child)': {
    marginBottom: variables.gap.small * 0.5,
  },
};

const info = css({
  ...messageBase,
  background: colors.accent,
}).toString();

const error = css({
  ...messageBase,
  background: colors.error,
}).toString();

const warning = css({
  ...messageBase,
  background: colors.warning,
}).toString();

export default {
  container,
  info,
  error,
  warning,
};
