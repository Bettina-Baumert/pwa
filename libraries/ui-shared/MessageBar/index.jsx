import React from 'react';
import PropTypes from 'prop-types';
import styles from './style';

/**
 * The MessageBar component.
 * @param {Object} props The component props.
 * @param {Array} props.messages The message content.
 * @param {Object} props.classNames Styling.
 * @return {JSX}
 */
const MessageBar = ({ messages, classNames }) => (
  <div className={`${classNames.container} ${styles.container}`}>
    {messages.map(({ type = 'info', message }) => (
      <div key={`${type}-${message}`} className={`${classNames.message} ${styles[type]}`}>
        {message}
      </div>
    ))}
  </div>
);

MessageBar.propTypes = {
  messages: PropTypes.arrayOf(PropTypes.shape({
    message: PropTypes.string,
    type: PropTypes.string,
  })).isRequired,
  classNames: PropTypes.shape({
    container: PropTypes.string,
    message: PropTypes.string,
  }),
};

MessageBar.defaultProps = {
  classNames: {
    container: '',
    message: '',
  },
};

export default MessageBar;
