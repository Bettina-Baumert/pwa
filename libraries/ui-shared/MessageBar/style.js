import { css } from 'glamor';
import { themeConfig } from '@shopgate/pwa-common/helpers/config';

const container = css({
  display: 'flex',
  flexDirection: 'column',
  background: themeConfig.colors.background,
}).toString();

const messageBase = {
  padding: `${themeConfig.variables.gap.small}px ${themeConfig.variables.gap.big}px`,
  fontSize: '0.875rem',
  fontWeight: 500,
  ':not(:last-child)': {
    marginBottom: themeConfig.variables.gap.small * 0.5,
  },
};

const info = css({
  ...messageBase,
  background: themeConfig.colors.accent,
  color: themeConfig.colors.accentContrast,
}).toString();

const error = css({
  ...messageBase,
  background: themeConfig.colors.error,
  color: themeConfig.colors.light,
}).toString();

const warning = css({
  ...messageBase,
  background: themeConfig.colors.warning,
  color: themeConfig.colors.light,
}).toString();

export default {
  container,
  info,
  error,
  warning,
};
