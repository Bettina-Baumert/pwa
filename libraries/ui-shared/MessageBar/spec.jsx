import React from 'react';
import { mount } from 'enzyme';
import MessageBar from './index';

const MESSAGE1 = 'This is some information';
const MESSAGE2 = 'This is an error that happened here.';
const MESSAGE3 = 'This is just a warning. Nothing to freak out about.';

describe('<MessageBar />', () => {
  describe('General rendering', () => {
    it('should be empty if no messages have been set', () => {
      const wrapper = mount(<MessageBar messages={[]} />);
      expect(wrapper).toMatchSnapshot();
    });

    it('should render a message as info if type is missing', () => {
      const wrapper = mount(<MessageBar messages={[{ message: 'something' }]} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Multiple messages rendering', () => {
    const wrapper = mount(<MessageBar
      messages={[
          {
            message: MESSAGE1,
          },
          {
            type: 'error',
            message: MESSAGE2,
          },
          {
            type: 'warning',
            message: MESSAGE3,
          },
        ]}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
