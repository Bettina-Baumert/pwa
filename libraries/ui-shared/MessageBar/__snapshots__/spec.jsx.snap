// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<MessageBar
  classNames={
    Object {
      "container": "",
      "message": "",
    }
  }
  messages={
    Array [
      Object {
        "message": "This is some information",
      },
      Object {
        "message": "This is an error that happened here.",
        "type": "error",
      },
      Object {
        "message": "This is just a warning. Nothing to freak out about.",
        "type": "warning",
      },
    ]
  }
>
  <div
    className=" css-1gmese"
  >
    <div
      className=" css-1psau2c"
      key="info-This is some information"
    >
      This is some information
    </div>
    <div
      className=" css-1t4ceri"
      key="error-This is an error that happened here."
    >
      This is an error that happened here.
    </div>
    <div
      className=" css-y06bhm"
      key="warning-This is just a warning. Nothing to freak out about."
    >
      This is just a warning. Nothing to freak out about.
    </div>
  </div>
</MessageBar>
`;

exports[`<MessageBar /> General rendering should be empty if no messages have been set 1`] = `
<MessageBar
  classNames={
    Object {
      "container": "",
      "message": "",
    }
  }
  messages={Array []}
>
  <div
    className=" css-1gmese"
  />
</MessageBar>
`;

exports[`<MessageBar /> General rendering should render a message as info if type is missing 1`] = `
<MessageBar
  classNames={
    Object {
      "container": "",
      "message": "",
    }
  }
  messages={
    Array [
      Object {
        "message": "something",
      },
    ]
  }
>
  <div
    className=" css-1gmese"
  >
    <div
      className=" css-1psau2c"
      key="info-something"
    >
      something
    </div>
  </div>
</MessageBar>
`;
